### PR TITLE
Favorites Drag and Drop Implementation

### PR DESCRIPTION
**Pull request by ehconitin**
Adds drag and drop functionality for favorites management, allowing users to:

- Move favorites between folders
- Move favorites from folders to orphan section
- Move orphan favorites into folders

Known Issues:
Drop detection at folder boundaries requires spacing workaround

### CODE REVIEW

File: packages/twenty-front/src/modules/favorites/components/FavoritesDroppable.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+}>`
+  position: relative;
+  transition: all 150ms ease-in-out;
+  width: 100%;
+
+  ${({ isDraggingOver, isDragIndicatorVisible, showDropLine, theme }) =>
+    isDraggingOver &&
+    isDragIndicatorVisible &&
+    `
+        background-color: ${theme.color.blueAccent20};
```
[the remark this answers is not in the document]
Comment (ehconitin) [reply packages/twenty-front/src/modules/favorites/components/FavoritesDroppable.tsx:24]: @Bonapara blueAccent20 looks awful in dark mode! Any suggestions?

File: packages/twenty-front/src/modules/favorites/components/CurrentWorkspaceMemberFavorites.tsx [hunk 2]
```diff
@@ -139,22 +138,31 @@ export const CurrentWorkspaceMemberFavorites = ({
...
+          <Droppable droppableId={`folder-${folder.folderId}`}>
+            {(provided) => (
+              <div
+                ref={provided.innerRef}
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...provided.droppableProps}
+                style={{
+                  marginBottom: 15,
+                }}
+                // TODO: (Drag Drop Bug) Adding bottom margin to ensure drag-to-last-position works. Need to find better solution that doesn't affect spacing.
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/favorites/components/CurrentWorkspaceMemberFavorites.tsx:163]: @charlesBochet 🫠, do you have inputs?

File: packages/twenty-front/src/modules/favorites/hooks/useReorderFavorite.ts [hunk 3]
```diff
@@ -11,28 +13,106 @@ export const useReorderFavorite = () => {
...
+        },
+      });
+
+      setActiveFavoriteFolderId(targetFolderId);
+      return;
+    }
+
+    if (destination.droppableId !== source.droppableId) {
+      const newFolderId =
+        destination.droppableId === 'orphan-favorites'
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/favorites/hooks/useReorderFavorite.ts:65]: can we introduce a const to materialize this orphan-favorites?

File: packages/twenty-front/src/modules/favorites/hooks/useReorderFavorite.ts [hunk 4]
```diff
@@ -11,28 +13,106 @@ export const useReorderFavorite = () => {
...
+    ) {
+      return;
+    }
+
+    const draggedFavorite = favorites.find((f) => f.id === draggableId);
     if (!draggedFavorite) return;
 
-    const inSameFolderFavorites = favoritesSorted.filter(
-      (fav) => fav.favoriteFolderId === draggedFavorite.favoriteFolderId,
+    if (destination.droppableId.startsWith('folder-header-')) {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/favorites/hooks/useReorderFavorite.ts:35]: please add tests on this hook
the naming of the hook should reflect that is is used for Drag and Drop as we are maniuplating droppable (and we have strong hypothesis on how the droppableId should be formed)

Comment (charlesBochet) [reply packages/twenty-front/src/modules/favorites/hooks/useReorderFavorite.ts:35]: can we also make this function throw in the case the droppableId is malformed

File: packages/twenty-front/src/modules/favorites/hooks/useReorderFavorite.ts [hunk 5]
```diff
@@ -11,28 +13,106 @@ export const useReorderFavorite = () => {
...
+        destination.droppableId === 'orphan-favorites'
+          ? null
+          : destination.droppableId.replace('folder-', '');
+
+      const destinationFavorites = favoritesSorted.filter(
+        (favorite) => favorite.favoriteFolderId === newFolderId,
+      );
+
+      let newPosition;
+      if (destinationFavorites.length === 0) {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/favorites/hooks/useReorderFavorite.ts:74]: this position things is already present in getDraggedRecordPosition, can we re-use it?

**Review by charlesBochet — COMMENTED** [review]
Overall great work! I've left comments on the code

**Comment by ehconitin** [comment]
@charlesBochet Tests related to this PR are done!
I am still not sure how we get around this issue https://github.com/twentyhq/twenty/pull/8979/files#diff-43b355cf0cd133b768296643d0b7499f08dde8de0adcce1bf704ac7129742337R164-R165

This is regarding the extra padding below each folder. Do you have any ideas?

File: packages/twenty-front/src/modules/favorites/components/FavoritesDroppable.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ehconitin) [reply packages/twenty-front/src/modules/favorites/components/FavoritesDroppable.tsx:24]: Much better! Thanks ;)

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**